packages/core/src/lib/Slider/Slider.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState, useEffect } from 'react';
+import { ISliderProps } from './types';
+import styles from './Slider.styles';
+
+const { DivStyled, LabelStyled, PStyled, InputStyled } = styles;
+
+const Slider: React.FC<ISliderProps> = ({
+    rangeId,
+    rangeLabel,
+    rangeMax,
+    rangeMin,
+    rangeValue,
+}) => {
+    const [currentValue, setCurrentValue] = useState(rangeValue);
+
```

Comment:
Let's not have an internal state instead let user pass a state variable. rangeValue could be used for that.
And then it also needs an onChange callback where user will provide the `setState` 
Something like below would be helpful.. 
```
const Comp = () =>{
  const [range,setRange] = useState(0);
  return <Slider value={range} onChange={(e)=>setRange(e.target.value)/>}
}
```

Reply:
will do

Reply:
@AbhinavMV the currentValue is for the input slider to understand the currentvalue, although i have added onChange callback for working on the same and getting back the value they need

Reply:
@3scava1i3r so I understand what you are saying but this is adding additional complexity which is not needed...(We have this problem in many components)

To understand the problem with this do the following things:
1. Go to storybook Forms/Slider/Default.
2. Now change the rangeValue prop from the storybook... you will notice that even after changing the value in storybook the component has not been updated.

Now the solution to this is simple. In the component, we can add-
```
useEffect(()=>{setCurrentValue(rangeValue ?? 0)},[rangeValue])
```

But you see the complexity increasing right... just because we are maintaining an internal state.
So a better approach would be to completely remove the internal state and let users control it themselves.
Hope I was able to explain properly. Feel free to ping me if still anything is unclear 😄